### ISSUE TITLE
Allow modifying staged files while using git full editor

### ISSUE THREAD

**Issue by jibbers42** [microsoft/vscode#160558]
I like to review my files in detail from SCM panel while creating the git message. Often during this process I find very minor issues that I like to change; things like: rename variable, modify comment, minor refactor, etc. It's easy enough to make these changes and run the test suite while still in the git full editor experience. The problem is SCM doesn't refresh nor include changes that are made.

Workflow:
- stage changes
- press SCM check mark to start git full editor
- begin editing message while reviewing diffs
- make small change, save file, run test suite
- notice SCM panel did not refresh to show unstaged changes, press SCM refresh button
- changed file is now in SCM unstaged changes area, click add button
- change seems to be staged
- finish git message, save and close git full editor
- changes that were staged when starting commit are committed, SCM refreshes to show newly staged changes still there and not commited

Specific changes for above workflow:
- SCM panel should refresh as usual during git full editor experience.
- Newly staged changes should be included in git commit when closing git full editor.

As an example, today I made a refactor and during review I noticed a section where I no longer needed a cast. After making the change, manually refreshing SCM, staging the file, finishing my commit message, and finally closing the git full editor, the SCM refreshed to show my change remained behind and was not committed.
![image](https://user-images.githubusercontent.com/797824/189439243-236b50ac-58a0-40f0-bbd9-e37bc79a02c1.png)

I suppose this could be avoided by reviewing the files in detail before starting git commit, but if you create detailed git messages, then you'd have to basically review the files in detail a second time to fill out the git message.

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by VSCodeTriageBot** [comment]
This feature request has not yet received the 20 community [upvotes](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) it takes to make to our backlog. 10 days to go. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by VSCodeTriageBot** [comment]
:slightly_frowning_face: In the last 60 days, this feature request has received less than 20 community upvotes and we closed it. Still a big Thank You to you for taking the time to create this issue! To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!
